packages/http/src/validate.ts

```diff
@@ -1,7 +1,66 @@
-import { Program } from "@typespec/compiler";
+import { Operation, Program, navigateProgram } from "@typespec/compiler";
 import { reportDiagnostic } from "./lib.js";
 import { getAllHttpServices } from "./operations.js";
-import { isSharedRoute } from "./route.js";
+import { getRoutePath, isSharedRoute } from "./route.js";
+import { OperationContainer } from "./types.js";
+
+function checkOperationReferenceContainerRoutes(program: Program) {
```

Comment:
I believe this should be a linter rule warning. `$onValidate` is typically reserved for errors.

Reply:
@catalinaperalta can you confirm?

Reply:
OK, I've looked at @catalinaperalta's changes to migrate the rules in `typespec-azure` and it makes sense, but I wonder if we start exposing linter rules from `@typespec/http` if we have to add an `extends` for it in `typespec-azure-core`.  What do you think @catalinaperalta?

Reply:
Updated the design to use the new linting framework instead. @timotheeguerin, how does it look?  Also, can you answer the question I asked above?

Reply:
@daviwil Yes we'd need to extend this in typespec-azure-core to keep checking for the warnings coming from `@typespec/http`

Reply:
Thanks!